packages/flutter_tools/lib/src/preview_device.dart

```diff
@@ -0,0 +1,185 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+// @dart = 2.8
+
+import 'dart:async';
+
+import 'package:meta/meta.dart';
+import 'package:process/process.dart';
+
+import 'application_package.dart';
+import 'base/file_system.dart';
+import 'base/io.dart';
+import 'base/logger.dart';
+import 'base/platform.dart';
+import 'build_info.dart';
+import 'bundle_builder.dart';
+import 'cache.dart';
+import 'desktop_device.dart';
+import 'devfs.dart';
+import 'device.dart';
+import 'device_port_forwarder.dart';
+import 'project.dart';
+import 'protocol_discovery.dart';
+
+typedef BundleBuilderFactory = BundleBuilder Function();
+
+BundleBuilder _defaultBundleBuilder() {
+  return BundleBuilder();
+}
+
+class PreviewDevice extends Device {
```

Comment:
Can you add a doc comment giving a general overview of what this is for?

Reply:
done